src/ExperimentalCycles.mo

```diff
@@ -0,0 +1,61 @@
+/// Cycles
+
+/// Usage of the Internet Computer is measured, and paid for, in _cycles_.
+/// This library provides imperative operations for observing cycles, transferring cycles and
+/// observing refunds of cycles.
+
+
+/// **WARNING:** This low-level API is **experimental** and likely to change or even disappear.
+/// Dedicated syntactic support for manipulating cycles may be added to the language in future, obsoleting this library.
+///
+/// **NOTE:** Since cycles measure computational resources, the value of
+/// `balance()` can change from one call to the next.
+
+import Prim "mo:prim";
+module {
+
+  /// Returns the actor's current balance of cycles as `amount`.
+  public func balance() : (amount : Nat) {
+    Prim.nat64ToNat(Prim.cyclesBalance())
+  };
+
+  /// Returns the currently available `amount` of cycles.
+  /// The amount available is the amount received in the current call,
+  /// minus the cumulative amount `accept`ed by this call.
+  /// On exit from the current shared function or async expression via `return` or `throw`
+  /// any remaining available amount is automatically
+  /// refunded to the caller/context.
+  public func available() : (amount : Nat) {
+    Prim.nat64ToNat(Prim.cyclesAvailable())
+  };
+
+  /// Transfers up to `amount` from `available()` to `balance()`
+  /// Returns the amount actually transferred, which may be less than 
+  /// requested, e.g. if less is available, or if canister balance limits are reached
+  public func accept(amount : Nat) : (accepted : Nat) {
+    Prim.nat64ToNat(Prim.cyclesAccept(Prim.natToNat64(amount)));
```

Comment:
Sorry for noting earlier, but shouldn't this consequently return 0 then if `amount` overflows the Nat64 range?

Reply:
Or, alternatively, pass 2^64-1 to the system and return whatever was accepted